superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx

```diff
@@ -88,24 +94,28 @@ const ModalFooter = ({ formData, closeModal }: ModalFooterProps) => {
 
 interface DrillByModalProps {
   column?: Column;
+  dataset: Dataset;
   filters?: BinaryQueryObjectFilterClause[];
   formData: BaseFormData & { [key: string]: any };
   groupbyFieldName?: string;
   onHideModal: () => void;
-  showModal: boolean;
-  dataset: Dataset;
 }
 
 export default function DrillByModal({
   column,
+  dataset,
   filters,
   formData,
   groupbyFieldName = 'groupby',
   onHideModal,
-  showModal,
-  dataset,
 }: DrillByModalProps) {
   const theme = useTheme();
+  const [chartDataResult, setChartDataResult] = useState<QueryData[]>();
+  const [drillBy, setDrillBy] = useState<DrillByType>(DrillByType.Chart);
```

Comment:
Could you change the name to sth more descriptive? Like for example `displayMode` or `drillByDisplayMode`?